src/GoogleFit/FitOnFhir.GoogleFit/Services/GoogleFitImportService.cs

```diff
@@ -97,6 +109,11 @@ public class GoogleFitImportService : ParallelTaskWorker<GoogleFitImportOptions>
                             await _eventHubProducerClient.SendAsync(eventBatch, cancellationToken);
                         }
                         while (pageToken != null);
+
+                        // Update the GoogleFitUser timestamp for this data source
+                        user.SyncTimes.Remove(dataStreamId);
+                        user.SyncTimes.Add(dataStreamId, currentTime);
+                        await _googleFitUserRepository.Update(user, cancellationToken);
```

Comment:
We are going to need to do something here to manage the multi-threaded nature of this process.

We are going to run into errors here because we fetch the user at the top of this method, but then update it here. If (when) another process updates this dictionary from another thread, the version of this entity will change, and the upsert will fail. The answer might be to collect all of the sync times and save them one time at the end of the sync, or handling a failure by fetching the user again, updating and trying the save again.